scripts/ci/check-consensus-release.sh

```diff
@@ -9,59 +9,59 @@
 # If $CI is set, then use `git show` as GHA does a sparse checkout. Otherwise,
 # do a normal cat of the cabal file
 
-cardano_packages=$(find . -maxdepth 1 -type d -name "ouroboros-consensus-cardano*" -or -name "ouroboros-consensus-shelley*" -or -name "ouroboros-consensus-byron*")
-cardano_last_version=$(grep "<a id=" ouroboros-consensus-cardano/CHANGELOG.md  | cut -d\' -f2 | cut -d- -f2 | head -n1)
-cardano_new_versions=$(if [[ -n "${CI}" ]]; then
-                           for f in $cardano_packages; do
-                               git show $(ls $f/*.cabal) | grep "+version" | rev | cut -d' ' -f1 | rev
-                           done
-                       else for f in $cardano_packages; do
-                               cat $f/*.cabal | grep "^version" | rev | cut -d' ' -f1 | rev
-                            done
-                       fi)
+function pkgs_in_bundle {
+  sed '/Changelog entries/q' $1/CHANGELOG.md | grep "\- \`ouroboros-" | cut -d\` -f2 | sort | uniq
+}
 
-consensus_packages=$(find . -maxdepth 1 -type d -name "ouroboros-consensus*" -not -name "*byron*" -not -name "*shelley*" -not -name "*cardano*")
-consensus_last_version=$(grep "<a id=" ouroboros-consensus/CHANGELOG.md  | cut -d\' -f2 | cut -d- -f2 | head -n1)
-consensus_new_versions=$(if [[ -n "${CI}" ]]; then
-                           for f in $consensus_packages; do
-                               git show $(ls $f/*.cabal) | grep "+version" | rev | cut -d' ' -f1 | rev
-                           done
-                       else for f in $consensus_packages; do
-                               cat $f/*.cabal | grep "^version" | rev | cut -d' ' -f1 | rev
-                            done
-                       fi)
+function get_last_version {
+  ch_version=$(grep "<a id=" $1/CHANGELOG.md  | cut -d\' -f2 | cut -d- -f2 | head -n1)
+  if [[ -z $ch_version ]]; then
+      if [[ $1 == "ouroboros-consensus-test" ]]; then
+          echo "0.3.1.0"
```

Comment:
This only happens if checking for the version of `ouroboros-consensus-test` for the first time when the changelog doesn't have any entries. Once there is one entry in the changelog, then this will be dead code, and it will indeed be dynamically computed